core/src/main/java/com/javadiscord/jdi/core/Discord.java

```diff
@@ -152,6 +156,27 @@ public void start() {
         EXECUTOR.execute(discordRequestDispatcher);
     }
 
+    public void stop() {
+        if (this.webSocketManager != null) {
+            this.webSocketManager.stop();
+        }
+
+        ThreadPoolExecutor threadPool = (ThreadPoolExecutor) EXECUTOR;
+        if (threadPool.getPoolSize() > 0) threadPool.shutdown();
+
+        try {
+            if (!threadPool.awaitTermination(30, TimeUnit.SECONDS)) {
+                threadPool.shutdownNow();
+                if (!threadPool.awaitTermination(30, TimeUnit.SECONDS)) {
+                    LOGGER.error("Threads not terminated properly");
+                }
+            }
+        } catch (InterruptedException ie) {
+            threadPool.shutdownNow();
+            Thread.currentThread().interrupt();
```

Comment:
y interrupt the main thread?

Also, log the error that's crucial

Reply:
That line is needed so that if e.g. `.awaitTermination` throws the `InterruptedException` due to termination gets interrupted somehow to maintain the status "interrupted" in the Thread. Since it could be that the Exception deletes the interrupted status of the Thread. I would not recommend to delete this line. But of course the log for the error is needed. Will add that

Reply:
[8fb4750](https://github.com/javadiscord/java-discord-api/pull/100/commits/8fb4750409fb9b008f47dbb2b1582f89c0befef1)

Reply:
This is good - `Thread.currentThread().interrupt();` is fine to do in this case. However, re-running the `threadPool.shutdownNow();` isn't - if it's failed, there is no need to keep invoking it multiple times.